hy/completer.py

```diff
@@ -83,29 +88,38 @@ def attr_matches(self, text):
         except Exception:
             return []
 
-        # Build match list to return
         n = len(attr)
         matches = []
         for w in words:
             if w[:n] == attr:
-                matches.append("%s.%s" % (expr, w.replace("_", "-")))
+                matches.append("{}.{}".format(
+                    expr.replace("_", "-"), w.replace("_", "-")))
         return matches
 
     def global_matches(self, text):
-        path = PATH
-        if self.namespace:
-            path.append(self.namespace)
+        matches = []
+        for p in self.path:
+            for k in p.keys():
+                if isinstance(k, str) or isinstance(k, six.text_type):
```

Comment:
If you want to check for both str and unicode and to be commpatible with Python 2 and Python 3, you can use `isinstance(k, six.string_types)`.